core/src/main/scala/org/apache/spark/serializer/SerializationDebugger.scala

```diff
@@ -111,7 +112,12 @@ private[spark] object SerializationDebugger extends Logging {
             visitExternalizable(e, elem :: stack)
 
           case s: Object with java.io.Serializable =>
-            val elem = s"object (class ${s.getClass.getName}, $s)"
+            val str = try {
+              String.valueOf(s)
+            } catch {
+              case _: SparkRuntimeException => "cannot print object"
+            }
+            val elem = s"object (class ${s.getClass.getName}, $str)"
```

Comment:
Is it possible to add a new test case? Or which existing test case covers this scenario?

Reply:
Add test `SPARK-51691 improveException swallow underlying exception` to cover this scenario

Reply:
can we just add a branch like `case s: Object with java.io.Serializable if Utils.isTesting =>`